http/codegen/service_data.go

```diff
@@ -1539,7 +1548,7 @@ func buildResponses(e *expr.HTTPEndpointExpr, result *expr.AttributeExpr, viewed
 						break
 					}
 				}
-				if needInit(result.Type) {
+				if needInit(result.Type) && !(e.SkipResponseBodyEncodeDecode && resp.HasBodyOnly()) {
```

Comment:
Are there cases where we call `needInit` without testing for `SkipResponseBodyEncodeDecode`? If not maybe `needInit` could be refactored to do that test as well?